src/components/Sidebars/Payback.js

```diff
@@ -19,11 +66,55 @@ const Payback = ({ cdp, reset }) => {
   const [liquidationPrice, setLiquidationPrice] = useState(0);
   const [collateralizationRatio, setCollateralizationRatio] = useState(0);
 
+  const [
+    { initialised, loaderToggle, allowanceSet, userHasAllowance },
+    dispatch
+  ] = useReducer(reducer, initialState);
+
+  useEffect(() => {
+    (async () => {
+      await checkAllowance();
+    })();
+  }, []);
+
   maker
     .getToken('MDAI')
     .balance()
     .then(daiBalance => setDaiBalance(daiBalance.toNumber()));
 
+  const getProxyToken = async () => {
+    const proxyAddress = await maker.service('proxy').ensureProxy();
```

Comment:
the proxy address seems like a great candidate for component state. maybe we could set it in the `useEffect` right below `checkAllowance`

Reply:
Will change